app/screens/settings-screen/settings/account-ln-address.tsx

```diff
@@ -0,0 +1,70 @@
+import { useState } from "react"
+import Clipboard from "@react-native-clipboard/clipboard"
+
+import { SettingsRow } from "../row"
+import { SetLightningAddressModal } from "@app/components/set-lightning-address-modal"
+
+import { useAppConfig } from "@app/hooks"
+import { useI18nContext } from "@app/i18n/i18n-react"
+
+import { gql } from "@apollo/client"
+import { useAccountLnAddressQuery } from "@app/graphql/generated"
+import { toastShow } from "@app/utils/toast"
+
+gql`
+  query AccountLNAddress {
+    me {
+      username
+    }
+  }
```

Comment:
have you look into how the network queries behave with this break down?

if `fetchPolicy` is not set, I believe it should be "cache-first" by default, but would not trigger query, but if it's not populated, it might send too many query to the server. 

not sure what would be the right solution. this is where relay from facebook shines

you should also ensure that mutation properties are been refetched from the server, because the default behavior from apollo client would be to show the cache value if available.

Reply:
this has now been added. thanks for the pointer